tests/resources/test_material_run.py

```diff
@@ -165,31 +165,38 @@ def test_filter_by_name(collection, session):
     assert sample_run['uids'] == runs[0].uids
 
 
-def test_list_by_name(collection, session):
+def test_cursor_paginated_searches(collection, session):
+    """
+    Tests that search methods using cursor-pagination are hooked up correctly.
+    There is no real search logic tested here.
+    """
     all_runs = [
         MaterialRunDataFactory(name="foo_{}".format(i)) for i in range(20)
     ]
     fake_request = make_fake_cursor_request_function(all_runs)
-    # lotta shady stuff going on in this test
+    # pretty shady, need to add these methods to the fake session to test their
+    # interactions with the actual search methods
     setattr(session, 'get_resource', fake_request)
+    setattr(session, 'post_resource', fake_request)
     setattr(session, 'cursor_paged_resource', Session.cursor_paged_resource)
-    # TODO: the results appear equal in the diff but fail equality check
+
     assert len(list(collection.list_by_name('unused', per_page=2))) == len(all_runs)
+    assert len(list(collection.list_all(per_page=2))) == len(all_runs)
+    assert len(list(collection.list_by_tag('unused', per_page=2))) == len(all_runs)
+    assert len(list(collection.list_by_attribute_bounds(
+        {LinkByUIDFactory(): IntegerBounds(1, 5)}, per_page=2))) == len(all_runs)
```

Comment:
Is this just testing that all of these methods (list by name, list all, list by tag, and list by attribute bounds) call get_resource or post_resource? Or is it testing something else?

Reply:
Pretty much. It ensures that they perform a valid invocation of `cursor_paged_resource`, which is not a **total** no-op test to do. I will admit that this is to a significant degree an 'appeasing 100% test coverage' test. The real testing will happen in nextgen-devkit.

Reply:
The test of `cursor_paged_resource` is more thorough: https://github.com/CitrineInformatics/citrine-python/blob/develop/tests/test_session.py#L189

Reply:
Well technically these tests call the same functions, actually. Let me say this: I think `cursor_paged_resource` is decently well tested, and these search functions don't really do anything except call `cursor_paged_resource`.

Reply:
Also gating 100% test coverage is dumb #hottakes

Reply:
😱
I think there's value in at least asserting that `list_by_tag` calls `cursor_paged_resource`. You could also check that it does so with the correct arguments.

Reply:
I actually think asserting that a particular function is called does more harm than good unless you're trying to test something around performance (e.g. request is only made once because it gets cached). I think that kind of testing is brittle and hard to maintain. With simple functions like these, in which you just call another function with arguments derived from user input and instance variables, tests devolve into 'test that this function is written the way I wrote it', which fails every time you change anything.
FWIW though, these tests do implicitly check that `cursor_paged_resource` is called, because the `len` assertions would not pass were the function not calling `cursor_paged_resource`.